app/models/account/credit_card.rb

```diff
@@ -134,18 +131,26 @@ def notify_credit_card_change
 
   private
 
+  delegate :payment_gateway_setting_exists?, to: :provider_account, prefix: true, allow_nil: true
+
   # voids the transaction with the +authorization+ code
   def void_transaction!(authorization)
     response = provider_payment_gateway.void(authorization)
     log_gateway_response(response, "void [transaction: #{authorization}]")
     response.success?
   end
 
+  def threescale_unstore_payment_gateway
+    return unless provider_account_payment_gateway_setting_exists?
```

Comment:
This is incorrect anyway. If the payment gateway setting of the provider is still there (and it will always be for master in Saas for example), you _need_ to unstore the credit card of the deleted account. It's more about trying to fetch the payment gateway of the provider account. If it exists, unstore the credit card; otherwise, just erase the payment details.